### DISCUSSION TITLE
Questions about observation function in ecloe

### DISCUSSION THREAD

**Discussion by superboom** [ds4dm/ml4co-competition#80] [Q&A] [answered]
Is there any API to extract number of specified constraint in ecole, like number of knapsack etc. ?

**Comment by gasse** [comment, marked as answer]
Hi @superboom ,

We don't have this in Ecole's API no. If that information is present in SCIP, I believe you should be able to extract it using the [PySCIPOpt API](https://github.com/scipopt/PySCIPOpt/blob/master/src/pyscipopt/scip.pyx). I know SCIP can somehow detect and classify constraints into types, but I am not sure how to access that information from the API.

@ambros-gleixner , @antoniach or @CGraczyk could maybe provide more guidance here ?

Best,
Maxime

**Reply by ambros-gleixner** [reply]
At C level yes, but this does not seem accessible through PySCIPOpt currently.
